ta/pkcs11/src/sanitize_object.c

```diff
@@ -313,11 +313,13 @@ enum pkcs11_rc sanitize_client_object(struct obj_attrs **dst, void *src,
 		    pkcs11_attr_is_boolean(cli_ref.id))
 			continue;
 
-		rc = sanitize_indirect_attr(dst, &cli_ref, data);
-		if (rc == PKCS11_CKR_OK)
-			continue;
-		if (rc != PKCS11_RV_NOT_FOUND)
-			return rc;
+		if (pkcs11_attr_has_indirect_attributes(cli_ref.id)) {
+			rc = sanitize_indirect_attr(dst, &cli_ref, data);
```

Comment:
we could change `sanitize_indirect_attr()` to return ERROR_GENRIC on invalid IDs, and remove this the test against `PKCS11_RV_NOT_FOUND` below.

Reply:
@vesajaaskelainen, what I was suggesting is illustrated in https://github.com/etienne-lms/optee_os/commit/4f4dddef109381dd36be313f15a1b1a59f3734a1. If you're fine with this added change please feel free to use it wihtout my S-o-b.

Reply:
Note  etienne-lms@4f4ddde referred right above needs some fixing (`#include <assert.h>` + s/cli_ref.id/cli_ref->id/) to be functionnal.

Reply:
I'll check this out for next round of updates.

Reply:
Merged to commits. Thanks.